tests/api_tests/test_add_documents.py

```diff
@@ -252,6 +262,134 @@ def test_add_documents_empty(self):
+                                            f"model_auth={quote_plus(json.dumps(model_auth))}",
+                                            f"mappings={quote_plus(json.dumps(mappings))}",
+                                            f"image_download_headers={quote_plus(json.dumps(image_download_headers))}"]
+
+        data = {
+            "documents": [
+                {
+                    "Title": "The Travels of Marco Polo",
+                    "Description": "A 13th-century travelogue describing the travels of Polo",
+                    "Genre": "History"
+                },
+                {
+                    "Title": "Extravehicular Mobility Unit (EMU)",
+                    "Description": "The EMU is a spacesuit that provides environmental protection",
+                    "_id": "article_591",
+                    "Genre": "Science"
+                }
+            ],
+            "nonTensorFields": ["Title", "Genre"]
+        }
+
+        for deprecated_query_parameters in deprecated_query_parameters_list:
+            url = f"{self.authorized_url}/indexes/{self.index_name_1}/documents?{deprecated_query_parameters}"
+            headers = {'Content-type': 'application/json'}
+
+            response = requests.post(url, headers=headers, data=json.dumps(data))
+            assert str(response.status_code).startswith("4")
+            self.assertIn("Marqo is not accepting any of the following parameters in the query string",
+                          str(response.json()))
+
+    def test_add_documents_extra_field(self):
+        """This test is to ensure that the new API does not accept extra body parameters"""
+        self.client.create_index(self.index_name_1)
+        url = f'{self.authorized_url}/{self.index_name_1}/documents'
+        headers = {'Content-type': 'application/json'}
+
+        data = {
+            "documents": [
+                {
+                    "Title": "The Travels of Marco Polo",
+                    "Description": "A 13th-century travelogue describing the travels of Polo",
+                    "Genre": "History"
+                },
+                {
+                    "Title": "Extravehicular Mobility Unit (EMU)",
+                    "Description": "The EMU is a spacesuit that provides environmental protection",
+                    "_id": "article_591",
+                    "Genre": "Science"
+                }
+            ],
+            "non_TensorFields": ["Title", "Genre"]  # not permitted field
+        }
+
+        response = requests.post(url, headers=headers, data=json.dumps(data))
+        assert str(response.status_code).startswith("4")
+        assert "Not Found" in str(response.json())
+
+    def test_add_documents_unknown_query_parameter(self):
+        """This test is to ensure that the new API does not accept unknown query parameters"""
+        self.client.create_index(self.index_name_1)
+        url = f'{self.authorized_url}/{self.index_name_1}/documents?rando_parameter=true'
```

Comment:
It would be better to have 1 test for bad query parameter and 1 test for bad body parameter. Since with the current test, we don't know what is causing the 4xx.